keps/sig-node/1029-ephemeral-storage-quotas/README.md

```diff
@@ -709,11 +734,161 @@ and are not reported here.
+  It can be calculated by `kubelet_volume_metric_collection_duration_seconds` metrics.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine
+the health of the service?**
+
+- [x] Metrics
+  - Metric name: `kubelet_volume_metric_collection_duration_seconds`
+  - Aggregation method: histogram
+  - Components exposing the metric: kubelet
+
+* **Are there any missing metrics that would be useful to have to improve observability of this feature? **
+
+  - Yes, there are no histogram metrics for each volume. The above metric was grouped by volume types because
+    the cost for every volume is too expensive.
+
+### Dependencies
+* **Does this feature depend on any specific services running in the cluster? **
+
+  -  Yes, the feature depneds on project quotas. Once quotas are enabled, the xfs_quota tool can be used to
+    set limits and report on disk usage.
+
+
+### Scalability
+* **Will enabling / using this feature result in any new API calls?**
+  - No.
+
+* **Will enabling / using this feature result in introducing new API types?**
+  - No.
+
+* **Will enabling / using this feature result in any new calls to the cloud
+provider?**
+  - No.
+
+* **Will enabling / using this feature result in increasing size or count of
+the existing API objects?**
+  - No.
+
+* **Will enabling / using this feature result in increasing time taken by any
+operations covered by [existing SLIs/SLOs]?**
+  - No.
+
+* **Will enabling / using this feature result in non-negligible increase of
+resource usage (CPU, RAM, disk, IO, ...) in any components?**
+  - Yes. It will use less CPU time and IO during ephemeral storage monitoring. `kubelet` now allows use of XFS quotas (on XFS and suitably configured ext4fs filesystems) to monitor storage consumption for ephemeral storage (currently for emptydir volumes only). This method of monitoring consumption is faster and more accurate than the old method of walking the filesystem tree. It does not enforce limits, only monitors consumption.
+
+### Troubleshooting
+
+<!--
+This section must be completed when targeting beta to a release.
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+-->
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+###### What are other known failure modes?
+
+1. If the ephemeral storage limitation is reached, the pod will be evicted by kubelet.
+
+2. It should skip when the image is not configured correctly (unsupported FS or quota not enabled).
+
+3. For "out of space" failure, kublet eviction should be triggered.
+
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+- Restart kubelet and wait for 1 minute to make the SLOs clear.(The volume stats checking interval is determined by kubelet flag `volumeStatsAggPeriod`(default 1m).)
```

Comment:
I don't see how this helps figure out *why* the SLO isn't being met. Is the recommendation to disable the feature in this case?

Reply:
I updated here to 
```
If the metrics show problems, we can check the log and quota dir with the below commands. 
- There will be warning logs([after the # is merged](https://github.com/kubernetes/kubernetes/pull/107490)) if volume calculation took too long than 1 second
- If quota is enabled, you can find the volume information and the processing time with `time repquota -P /var/lib/kubelet -s -v`
```

No, we recommend enabling this feature gate for performance.

I wanted to explain how to make the metric clear in the old message. Probably, this is not needed.
- The metrics `volume_metric_collection_duration_seconds` is a histogram metric that contains a lot of old data.
A restart may help you to clearly know what is the current status of volume monitoring. 
- `volumeStatsAggPeriod` flag is the time that we need to wait before checking the metrics.